src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -39,13 +40,11 @@ pub struct FlowServiceImpl {
     state: Arc<Mutex<State>>,
     run_config: Arc<FlowServiceRunConfig>,
     event_bus: Arc<EventBus>,
-    flow_event_store: Arc<dyn FlowEventStore>,
     time_source: Arc<dyn SystemTimeSource>,
-    task_scheduler: Arc<dyn TaskScheduler>,
-    flow_configuration_service: Arc<dyn FlowConfigurationService>,
     dataset_changes_service: Arc<dyn DatasetChangesService>,
     dependency_graph_service: Arc<dyn DependencyGraphService>,
     dataset_ownership_service: Arc<dyn DatasetOwnershipService>,
+    catalog: Catalog,
```

Comment:
This is a place we need to think over better - maybe later:
 - we have components that are transaction-independent (i.e. time source)
 - we have components that can only function with a transactional catalog (event store)
 - we have components that can control scope of the transaction (flow service).

But actually, some parts of flow service inject transaction-dependent components like event store, and now we are moving these from the component static state to function local vars / arguments. This indicates that we have an implicit logic boundary, and should try isolating sub-components to be able to use DI again (from transactional catalog). Those sub-components would be short-lived, with scope limited to the duration of 1 transaction.